File: training/cpp/src/de/sots/cellarsandcentaurs/domain/service/CreatureService.cpp
#include "CreatureService.hpp"

namespace de::sots::cellarsandcentaurs::domain::service {

CreatureService::CreatureService(std::shared_ptr<Creatures> creatures)
    : creatures_{std::move(creatures)} {}

void CreatureService::save(const model::Creature& creature) {
    creatures_->save(creature);
}

model::Creature CreatureService::find(const model::CreatureId& id) {
    return creatures_->find(id);
}

}

File: sample-projects/vue/src/de/sots/cellarsandcentaurs/domain/service/CreatureService.ts
import log from 'loglevel'
import { Creatures } from './Creatures'
import { Creature } from '../model/Creature'
import { CreatureId } from '../model/CreatureId'

export class CreatureService {
  private lastWalkingSpeed?: import('../model/Speed').Speed

  constructor(private readonly creatures: Creatures) {}

  async save(creature: Creature): Promise<void> {
    log.info('saving creature ' + creature.getId().id)
    this.lastWalkingSpeed = creature.getSpeeds().get(SpeedTypeWalking)
    await this.creatures.save(creature)
  }

  find(id: CreatureId): Promise<Creature> {
    return this.creatures.find(id)
  }
}

const SpeedTypeWalking = 'walking' as import('../model/SpeedType').SpeedType
